### PR TITLE
feat: [MSSQL] expose storage encryption

### PR DESCRIPTION
**Pull request by fnaranjo-vmw**
[#185149632]

### Checklist:

* [x] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

**Review by pivotal-marcela-campo — CHANGES_REQUESTED** [review]

File: terraform-tests/mssql_test.go [hunk 1]
```diff
@@ -261,4 +265,52 @@ var _ = Describe("mssql", Label("mssql-terraform"), Ordered, func() {
...
+		When("a kms_key_id is passed and storage_encrypted is false", func() {
+			It("should complain about kms_key_id and storage_encrypted mismatch - storage_encrypted: false", func() {
+				session, _ := FailPlan(terraformProvisionDir, buildVars(defaultVars, requiredVars, map[string]any{"kms_key_id": "some-kms-id", "storage_encrypted": false}))
+
+				Expect(session.ExitCode()).NotTo(Equal(0))
+				Expect(session).To(gbytes.Say("set `storage_encrypted` to `true` or leave `kms_key_id` field blank"))
+			})
+		})
+
+		When("an invalid kms_key_id is passed and storage_encrypted is true", func() {
```
[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply terraform-tests/mssql_test.go:299]: Yes! Great catch. Thanks.

File: terraform/mssql/provision/validations.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+resource "terraform_data" "kms_key_was_not_provided" {
```
[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply terraform/mssql/provision/validations.tf:1]: You are absolutely right.
Removing. Great catch.

Thanks for noticing.
I wasn't aware or completely forgot about such default behaviour.
I agree forcing customers to specify the AWS managed key ARN is not nice.

File: terraform-tests/mssql_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply terraform-tests/mssql_test.go:299]: Sorry, nop. It is a legitimate test with wrong copy-pasted title.
I'll fix it.

[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply terraform-tests/mssql_test.go:297]: Fixed the description.

File: aws-mssql.yml [hunk 3]
```diff
@@ -103,6 +115,7 @@ provision:
     variables: terraform/mssql/provision/variables.tf
     main: terraform/mssql/provision/main.tf
     data: terraform/mssql/provision/data.tf
+    validations: terraform/mssql/provision/validations.tf
```
Comment (pivotal-marcela-campo) [inline aws-mssql.yml:118]: praise: love it

File: terraform-tests/mssql_test.go [hunk 4]
```diff
@@ -261,4 +265,50 @@ var _ = Describe("mssql", Label("mssql-terraform"), Ordered, func() {
...
+		When("with default values", func() {
+			It("should succeed", func() {
+				plan = ShowPlan(terraformProvisionDir, buildVars(defaultVars, requiredVars, nil))
+				Expect(ResourceCreationForType(plan, "aws_db_subnet_group")).To(HaveLen(1))
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(MatchKeys(IgnoreExtras, Keys{"storage_encrypted": BeTrue()}))
+			})
+		})
+
+		When("no kms_key_id passed and storage_encrypted set to true", func() {
+			// Currently, this tests the exact same inputs as the one `with default values`
```
Comment (pivotal-marcela-campo) [inline terraform-tests/mssql_test.go:279]: suggestion: Interesting take, maybe we can discuss in tech forum and see if we want to make this the standard.

**Review by pivotal-marcela-campo — APPROVED** [review]

**Pull request merged**